pkg/telemetry/prometheus_resourcer.go

```diff
@@ -25,8 +25,14 @@ type PrometheusResourcer struct {
 }
 
 func (cfg PrometheusResourcer) validate() error {
-	if cfg.MonitoringTargetNS == "" || cfg.MonitoringTargetName == "" || cfg.ServiceMonitorName == "" {
-		return TelemetryConfigIncomplete{}
+	if cfg.MonitoringTargetNS == "" {
```

Comment:
Not directly related to the issue at hand, but this was returning incomplete error messages and making my unit tests fail.